source/js/searcher.js

```diff
@@ -0,0 +1,771 @@
+ */
+function sendSearch(value, site, trimWhitespaces = true) {
+    const formattedData = F.getRequestDataFormatter(value, trimWhitespaces);
+    const formUrl = location.origin + location.pathname.split("/html/")[0] + `/html/${C.CURRENT_LANG}/${site}?search=${formattedData}`;
+    location.href = formUrl;
+}
+
+/**
+ * Method used for loading the paths of the available html files to be searched through.
+ * @returns JSON parsed object on success, otherwise null when the parsing of the response is incorrect, otherwise false if the `CURRENT_URL_SCHEME` is incorrect.
+ */
+async function loadWebsitePaths() {
+    if (!C.CURRENT_URL_SCHEME_CORRECT) return false;
+    return fetch(C.FILE_PATH).then(response => response.text()).then(text => {
+        try {
+            return JSON.parse(text);
+        } catch (e) {
+            console.error(e);
+        }
+    });
+}
+
+/**
+ * Method used for reading the content of the file provided by the path.
+ * @param {String} path The path to the file to be read.
+ * @returns The text from the file, otherwise, the reason the reading failed. Returns false in case the `CURRENT_URL_SCHEME` is incorrect.
+ */
+async function readFileText(path) {
+    if (!C.CURRENT_URL_SCHEME_CORRECT) return false;
+    return fetch(path).then(response => response.text());
+}
+
+/**
+ * Private method used for initializing the searcher API.
+ * @returns After awaiting `true` is returned when to loading was a success, otherwise, `false`.
+ */
+async function __initSearcher() {
+    return loadWebsitePaths().then((val) => {
+        if (!val || val instanceof String || val instanceof Boolean) {
+            //Error occured while fetching the json file containing the websites and languages available.
+            console.error("Searcher API cannot be loaded as an unexpected error occured while initializing.")
+            return false;
+        } else {
+
+            //Assign the available WEBSITES and LANGUAGES of the website.
+            WEBSITE_PATHS = val[0];
+            WEBSITE_LANGUAGES = val[1];
+
+            /**
+             * Method used for checking if the key can be found in the sessionStorage object.
+             * @param {String} key The key name in the sessionStorage object.
+             * @returns {Boolean} True when the key is found in the sessionStorage object, otherwise, false.
+             */
+            var testSessionStorageForKey = (key) => sessionStorage.getItem(key) != null;
+
+            /**
+             * Method used for determining whether all the languages are in the sessionStorage object.
+             * @returns The keys that are not contained in the sessionStorage object.
+             */
+            var getMissedKeys = () => {
+                let temp = WEBSITE_LANGUAGES;
+                return temp.filter((lang, i, args) => !testSessionStorageForKey(lang));
+            };
+
+            var loadDataToSessionStorage = async(langs, paths) => {
```

Comment:
this part is a bit hard to read - maybe add more explanation?

Reply:
Added comments for this whole function, as it can be remade into using indexedDB in the future, thus removing the need for JSON parsing and making the search process faster. With easier manipulation of the data already stored in the database.